WifiART/src/structs/CivicAddressTypes.java

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package structs;
+
+/**
+ * A wrapper class that provides data structures containing Civic Address Types (CATypes),
+ * as specified in RFC 4776.
+ */
+public class CivicAddressTypes {
+    private static final String SELECTION_PROMPT = "SELECT TYPE";
```

Comment:
I was viewing this class more as a wrapper for any data structures that include this set of data, so that those structures can be accessed by any class. My reasoning for adding the prompt to the list here is that this requires less steps compared to creating an array of just the values, then having to copy these values into a new array with increased size so the prompt can be included. Perhaps it would make more sense to return a 1-indexed array, leaving index 0 as an empty string, then simply changing the value at index 0 to the prompt so the prompt is defined in the class where it is used.

Reply:
After reviewing this further, it seems that using a Vector is the way to go, since Vectors allow for increased capacity. I can store a Vector in the CivicAddressTypes class, then insert the selection prompt to this Vector in ArtMvcView and pass in the Vector to the JComboBox.

Reply:
Vector is a legacy class and should not be used anymore. It adds synchronization when it's not necessary. Consider using ArrayList instead.

For more information - https://errorprone.info/bugpattern/JdkObsolete

Reply:
Java Swing still uses Vectors, and a JComboBox can be initialized from a Vector or array but not an ArrayList. My guess is that thread-safety was required in this case. An alternate solution would be to use an ArrayList rather than a Vector, and convert that to an array before passing it into the JComboBox.

Reply:
"If a synchronized collection is necessary, use Collections.synchronizedList or a data structure from java.util.concurrent."

I would prefer using ArraryList and convert it to Array.

Reply:
Yes, this seems more intuitive, even though it takes more steps compared to using a Vector.